web_client/src/textIntervalReader.jsx

```diff
@@ -0,0 +1,42 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
+/**
+ *  A higher order component that reads text from `readTextInterval`: an async function that returns
+ *  `{ text, readingEvery }`, where `text` is a text string read and `readingEvery` is the number
+ *  of seconds after which new `text` should be read. Every `readingEvery` seconds, this component
+ *  will read text from `readTextInterval` and pass the read text to `WrappedComponent`s `text` prop.
+ */
+const withTextIntervalReader = (WrappedComponent, readTextInterval) => (
+    class extends Component {
+        state = {
+            text: ''
+        }
+        onTextIntervalRead({ text }) {
```

Comment:
This is past tense. 'read' as in, we have 'read' the text interval. I just like `on` since it makes it clear it's an event handler.

I do realize it's a little confusing because 1) 'read' is ambiguous and 2) `onButtonClick` is the standard, usually starts with on and is a present tense verb. In my ideal world it's `onButtonClicked`.